substrate/frame/nomination-pools/src/lib.rs

```diff
@@ -3169,8 +3229,34 @@ impl<T: Config> Pallet<T> {
 				"depositor must always have MinCreateBond stake in the pool, except for when the \
 				pool is being destroyed and the depositor is the last member",
 			);
+
+			let expected_tvl: BalanceOf<T> = BondedPools::<T>::iter()
+				.map(|(id, inner)| {
+					T::Staking::total_stake(
+						&BondedPool { id, inner: inner.clone() }.bonded_account(),
+					)
+					.unwrap_or_default()
+				})
+				.reduce(|acc, total_balance| acc + total_balance)
+				.unwrap_or_default();
+
+			ensure!(
+				TotalValueLocked::<T>::get() == expected_tvl,
+				"TVL deviates from the actual sum of funds of all Pools."
+			);
+
+			let total_balance_members: BalanceOf<T> = PoolMembers::<T>::iter()
```

Comment:
Similarly for this. The total balance could be calculated when iterate over `PoolMembers` earlier in the function.